cmd/summary.go

```diff
@@ -233,17 +155,91 @@ func GetSummaryCommand() *cobra.Command {
 						}
 						return errors.Join(errs...)
 					}
-					return nil
 				}
+			} else {
+				// if an invalid number of arguments are provided, print usage
+				PrintHowToUse("summary")
+				return errors.New("invalid number of arguments provided")
 			}
-			pterm.Error.Println("wrong number of arguments, expecting two (2)")
 			return nil
 		},
 	}
 	cmd.Flags().BoolP("no-color", "n", false, "Disable color and style output (very useful for CI/CD)")
 	return cmd
 }
 
+func listenForUpdates(updateChan chan *model.ProgressUpdate, errorChan chan model.ProgressError, doneChan chan bool, failed bool) {
```

Comment:
The `failed` bool is never updated.

Currently - even if there's a `spinner.Fail` call, we mark  it as done and return nothing. No errors or anything like that. Is this desired behaviour?

Should a failure in one `listenForUpdates` call result in a failure in all other ones?
If so - should we use a `failed *bool` instead of `failed bool`, and set it to true wherever there's a `spinner.Fail` call?

Or should we instead utilize the `errorChan`? What is the difference between `failed` and `errorChan`?